apps/user/filters.py

```diff
@@ -46,6 +46,13 @@ def filter_exclude_usergroup(self, qs, name, value):
 
     def filter_search(self, qs, name, value):
         if value:
+            first_name, last_name = value.split(' ', 1) if ' ' in value else (value, '')
+        if first_name and last_name:
+            qs = qs.filter(
+                models.Q(first_name__icontains=first_name) &
+                models.Q(last_name__icontains=last_name)
+            )
+        else:
```

Comment:
Another approach can be to annotate "first_name last_name" at the query level and then use the value directly to search. This way if there are multiple words, it will also work.

https://docs.djangoproject.com/en/4.2/topics/db/aggregation/#generating-aggregates-for-each-item-in-a-queryset
https://docs.djangoproject.com/en/3.2/ref/models/database-functions/#concat

Also, let's update the test case with this new search logic.

Reply:
@thenav56 ,I have updated the function.
Can you please point me to where the test cases are ?

Reply:
Super. Let's also keep existing filters as well.  cc: @AdityaKhatri 
```
                models.Q(first_name__icontains=value) |
                models.Q(last_name__icontains=value) |
 ```

## Add/Update test
We normally add test cases in the individual app directory. For users it will be inside `apps/user/tests/`, since the changes are related to a **query**, we add test cases to `test_schemas.py` or (`test_queries.py`)
For the user's query, lucky we already have one which is defined here https://github.com/the-deep/server/blob/develop/apps/user/tests/test_schemas.py#L535. Now we will need to make sure that new cases are also tested here. 

![2023-07-05-094049](https://github.com/the-deep/server/assets/7059255/b17a25b3-b298-4271-b99c-e5ec48c3b231)